diff --git a/package.json b/package.json
index 7560a2e..4fb00d2 100644
--- a/package.json
+++ b/package.json
@@ -42,5 +42,10 @@
       "last 1 firefox version",
       "last 1 safari version"
     ]
+  },
+  "jest": {
+    "transformIgnorePatterns": [
+      "/node_modules/(?!(axios|react-day-picker)/)"
+    ]
   }
 }
diff --git a/src/App.test.js b/src/App.test.js
deleted file mode 100644
index 1f03afe..0000000
--- a/src/App.test.js
+++ /dev/null
@@ -1,8 +0,0 @@
-import { render, screen } from '@testing-library/react';
-import App from './App';
-
-test('renders learn react link', () => {
-  render(<App />);
-  const linkElement = screen.getByText(/learn react/i);
-  expect(linkElement).toBeInTheDocument();
-});
diff --git a/src/components/CodeEditorWindow/parser.test.js b/src/components/CodeEditorWindow/parser.test.js
new file mode 100644
index 0000000..5f14854
--- /dev/null
+++ b/src/components/CodeEditorWindow/parser.test.js
@@ -0,0 +1,39 @@
+import { SomeFunctionToExtractRequests } from './WHATEVER_THE_FUNCTION_IS_CALLED';
+
+
+let testCode = `
+
+GET collections
+
+PUT collections/demo1
+{
+    "vectors":
+    {
+        "size": 1,
+        "distance": "Cosine"
+    }
+}
+
+GET collections
+
+DELETE collections/demo1
+
+GET collections
+
+GET collections/startups
+
+`;
+
+
+
+describe('parser', () => {
+    it('Should extract query blocks from code', () => {
+
+        let requests = SomeFunctionToExtractRequests(testCode);
+
+        expect(requests.length).toEqual(6);
+
+
+    });
+});
+
